File: backend/services/SearchService/SearchService.API/Models/Course.cs
namespace SearchService.API.Models
{
    public record UserInCourse(string Id, string Name, string? ThumbnailUrl);
    public record CategoryInCourse(string Id, string Name, string Slug);
    public record Locale(string Name, string EnglishTitle);

    public class Course
    {
        public string Id { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string TitleCleaned { get; set; } = string.Empty;
        public string[] LearnedContents { get; set; } = [];
        public string Level { get; set; } = string.Empty;
        public bool IsPaid { get; set; }
        public decimal Price { get; set; }
        public decimal Rating { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int StudentCount { get; set; }
        public Locale? Locale { get; set; }
        public string? ThumbnailUrl { get; set; }
        public UserInCourse User { get; set; } = null!;
        public CategoryInCourse Category { get; set; } = null!;
        public CategoryInCourse SubCategory { get; set; } = null!;
        public string[] Tags { get; set; } = [];
    }
}

File: frontend/vicourses_admin_web/src/types/course.ts
import { VideoFile } from "./common";

export type CourseLevel = "All" | "Basic" | "Intermediate" | "Expert";
export type CourseStatus = "Unpublished" | "WaitingToVerify" | "Published";
export type LessonType = "Video" | "Quiz";

export type Course = {
    id: string;
    title: string;
    titleCleaned: string;
    learnedContents: string[];
    level: CourseLevel;
    status: CourseStatus;
    isApproved: boolean;
    isPaid: boolean;
    price: number;
    rating: number;
    createdAt: string;
    updatedAt: string;
    studentCount: number;
    locale: {
        name: string;
        englishTitle: string;
    } | null,
    thumbnailUrl: string | null;
    user: {
        id: string;
        name: string;
        thumbnailUrl: string | null;
    },
    category: {
        id: string;
        name: string;
        slug: string;
    },
    subCategory: {
        id: string;
        name: string;
        slug: string;
    },
    tags: string[],
}

export type CourseDetail = Course & {
    description: string | null;
    requirements: string[],
    targetStudents: string[],
    previewVideo: VideoFile | null,
    metrics: {
        sectionsCount: number;
        lessonsCount: number;
        quizLessonsCount: number;
        totalVideoDuration: number;
    },
}

export type InstructorCurriculum = {
    sections: SectionInInstructorCurriculum[];
}

export type SectionInInstructorCurriculum = {
    id: string;
    courseId: string;
    title: string;
    description: string | null;
    order: number;
    duration: number;
    lessonCount: number;
    lessons: Lesson[];
};

export type Quiz = {
    number: number;
    title: string;
    isMultiChoice: boolean;
    answers: {
        number: number;
        title: string;
        isCorrect: boolean;
        explanation: string | null;
    }[];
};

export type Lesson = {
    id: string;
    courseId: string;
    sectionId: string;
    userId: string;
    title: string;
    order: number;
    type: LessonType;
    createdAt: string;
    updatedAt: string;
    description: string | null;
    video: VideoFile | null;
    quizzesCount: number;
    quizzes: Quiz[];
};